ibm/mas_devops/roles/aiservice_tenant/tasks/tenant/remove/main.yml

```diff
@@ -9,42 +9,137 @@
-  shell: oc delete sa km-s3-sa -n {{ tenantNamespace }}
-  ignore_errors: true
-  register: delete_result
-  failed_when: delete_result.rc != 0 and "NotFound" not in delete_result.stderr
 
-- name: Delete km-s3-secret Secret
-  shell: oc delete secret km-s3-secret -n {{ tenantNamespace }}
-  ignore_errors: true
-  register: delete_result
-  failed_when: delete_result.rc != 0 and "NotFound" not in delete_result.stderr
+- name: Set facts from AIServiceTenant
+  ansible.builtin.set_fact:
+    aiservice_instance_id: "{{ lookup_result.resources[0].metadata.labels['app.kubernetes.io/instance'] | default('') }}"
+    aiservice_tenant_id: "{{ lookup_result.resources[0].spec.settings.tenant.tenantId | default('') }}"
+    aiservice_tenant_name: "{{ lookup_result.resources[0].metadata.name }}"
+  when: lookup_result.resources | length > 0
 
-- name: Delete regcred Secret
-  shell: oc delete secret regcred -n {{ tenantNamespace }}
-  ignore_errors: true
-  register: delete_result
-  failed_when: delete_result.rc != 0 and "NotFound" not in delete_result.stderr
+- name: Debug facts set from AIServiceTenant
+  debug:
+    msg:
+      - "aiservice_instance_id: {{ aiservice_instance_id }}"
+      - "aiservice_tenant_id: {{ aiservice_tenant_id }}"
+      - "aiservice_tenant_name: {{ aiservice_tenant_name }}"
 
-- name: Delete tenant namespace
-  shell: oc delete ns {{ tenantNamespace }}
-  ignore_errors: true
-  register: delete_result
-  failed_when: delete_result.rc != 0 and "NotFound" not in delete_result.stderr
+- name: Define resource kinds to delete
+  set_fact:
+    resource_kinds:
+      - ConfigMap
+      - Secret
+      - Route
+      - DeploymentConfig
+      - VolumeSnapshot
+      - Deployment
+      - Pod
+      - PersistentVolumeClaim
+      - StatefulSet
+      - Service
+      - NetworkPolicy
+
+- name: Define API version map for resource kinds
+  set_fact:
+    api_version_map:
+      ConfigMap: v1
+      Secret: v1
+      Route: route.openshift.io/v1
+      DeploymentConfig: apps.openshift.io/v1
+      VolumeSnapshot: snapshot.storage.k8s.io/v1beta1
+      Deployment: apps/v1
+      Pod: v1
+      PersistentVolumeClaim: v1
+      StatefulSet: apps/v1
+      Service: v1
+      NetworkPolicy: networking.k8s.io/v1
+
+- name: Gather all resources by kind in namespace
+  loop: "{{ resource_kinds }}"
+  loop_control:
+    loop_var: resource_kind
+  kubernetes.core.k8s_info:
+    api_version: "{{ api_version_map[resource_kind] }}"
+    kind: "{{ resource_kind }}"
+    namespace: "{{ tenantNamespace }}"
+  register: resource_list
+  ignore_errors: yes
+
+- name: Combine all gathered resources into one list
+  set_fact:
+    items_to_delete: >-
+      {{
+        resource_list.results
+        | map(attribute='resources')
+        | list
+        | flatten
+      }}
+
+- name: Debug gathered resources to delete
+  debug:
+    msg: "Found {{ items_to_delete | length }} resources to delete in namespace '{{ tenantNamespace }}'"
```

Comment:
why does the ansible role do this? it should be the aiservice tenant operator that removes these items.